app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java

```diff
@@ -148,12 +149,20 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.auth_ui_layout);
         ButterKnife.bind(this);
 
-        FirebaseAuth auth = FirebaseAuth.getInstance();
-        if (auth.getCurrentUser() != null) {
-            startSignedInActivity(null);
-            finish();
-            return;
-        }
+        new Handler().post(new Runnable() {
+            @Override
+            public void run() {
+                // We need to post this to the main thread to ensure `onActivityResult` is called
```

Comment:
Yeah, so here's the issue I'm trying to solve: when going to a deeply nested activity (such as being 3 activities deep in the `EmailActivity`), the `AuthUiActivity` is going to die. When it's recreated, `onActivityResult` gets called _after_ `onCreate` which is too late by then because we've already started killing the activity.  I'm just posting this to the main thread as a hack to ensure that the `onActivityResult` message gets delivered first. Does that make sense?

Reply:
What if we moved to `onResume()`, isn't that always called after `onActivityResult` and `onCreate` have both finished?

Reply:
Ohhhhhhhhh, after a super long time of Googling around and trying to understand what was happening, something clicked and I finally figured it out. Everybody says that `onActivityResult` is guaranteed to be called _before_ `onResume`, but that's never what I've observed. Here's the thing though: the system also guarantees that `onResume` will be called if there's only a transparent activity on top. And guess what? We have a transparent `KickoffActivity`!

Do you remember [all](https://github.com/firebase/FirebaseUI-Android/issues/975) [those](https://github.com/firebase/FirebaseUI-Android/issues/553) [issues](https://github.com/firebase/FirebaseUI-Android/issues/182) where people are like "`onActivityResult` isn't being called" and we're like "well, we know it is." Those people probably copied from the sample and are running into the same "bug!"

I need to see if there's an API to check if an activity is on top of you or not, but otherwise I can't think of a better solution than the post.

Reply:
I just ran the sample app and I see `onResume` come after `onActivityResult` in `AuthUiActivity` every time.

Reply:
@samtstern Oh lol, you know what? It's a bug with the developer option "Don't keep activities" I've been using to test this stuff. I can confirm that the expected order of callbacks is correct with that dev option disabled. Anyway, I'm going to report that to AOSP, but glad to know I'm not completely losing my mind.